doc/model.rst

```diff
@@ -0,0 +1,2354 @@
+
+To enable a SmartRedis ``Client`` to search a number of ``Model`` prefixes when requesting data from the ``Orchestrator``,
+execute ``Model.register_incoming_entity()`` on the designated ``Model``. The key components of this function include:
+
+- If a ``Model`` application requests prefixed tensors placed by another ``Model``, the interaction
+  between the two entities must be registered in the ``Experiment`` driver script. If the interaction is not
+  properly registered, ``Client.set_data_source()`` will not be able to find the ``Models`` prefix
+  that placed the tensors in the ``Orchestrator``.
+- This function enables users to execute ``Client`` functions such as ``Client.delete_tensor(tensor_name)`` on
+  prefixed tensors without having to specify the prefix. The prefix is instead set via ``Client.set_data_source(model_name)``
+  in the application and registered via ``Model.register_incoming_entity()`` in the ``Experiment`` driver script.
+
+.. _client_prefix_func:
+----------------
+Client functions
+----------------
+A ``Client`` object supports five prefixing functions: ``Client.use_tensor_ensemble_prefix()``,
+``Client.use_dataset_ensemble_prefix()``,  ``Client.use_list_ensemble_prefix()``,
+``Client.use_model_ensemble_prefix()`` and ``Client.set_data_source()``.
+
+To enable or disable SmartRedis data structure prefixing for tensors, ``Datasets``, aggregation lists, ML models
+and scripts, SmartRedis ``Client`` offers functions per data structure:
+
+- Tensor: ``Client.use_tensor_ensemble_prefix()``
+- ``Dataset``: ``Client.use_dataset_ensemble_prefix()``
+- Aggregation lists: ``Client.use_list_ensemble_prefix()``
+- ML models/scripts: ``Client.use_model_ensemble_prefix()``
+
+.. warning::
+    To access the ``Client`` prefixing functions, prefixing must be enabled on the
+    ``Model`` through ``Model.enable_key_prefixing()``. This function activates prefixing
+    for tensors, ``Datasets`` and lists.
+
+For examples on instructing SmartSim to send prefixed data to the ``Orchestrator``, navigate to the
+:ref:`put/set operations<put_set_prefix>` section.
+
+Users can instruct the SmartRedis ``Client`` to search a number of prefixes when requesting data from the ``Orchestrator``.
+This is achieved through the ``Client.set_data_source()`` function used within the ``Model``
+application. To implement this functionality:
+
+1. Use ``Model.register_incoming_entity()`` on the ``Model`` intending to search for data in the ``Orchestrator``
+   placed by a separate ``Model``.
+2. Pass the SmartSim entity (e.g., another ``Model``) to ``Model.register_incoming_entity()`` in order to
+   reference the ``Model`` prefix in the application code.
+3. In the ``Model`` application, instruct the ``Client`` to prepend the specified ``Model`` name during key searches
+   using ``Client.set_data_source("model_name")``.
+
+For examples on instructing a ``Client`` to append a ``Model`` name to a key when searching for data, read the
```

Comment:
Change this sentence to something more general like:

Examples are provided below that show the use of these client methods in conjunction with the SmartSim key prefixing Model API functions.